JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java

```diff
@@ -300,7 +380,38 @@ public void deleteTimeSeries(String tsIRI, Connection conn) {
 		// Extract "backup" information (dataIRIs, TimeUnit, DBUrl) for potential later re-instantiation (in case RDB deletion fails)
 		List<String> dataIRIs = rdfClient.getAssociatedData(tsIRI);
 		String timeUnit = rdfClient.getTimeUnit(tsIRI);  // can be null
-		
+
+		//Get time series type
+		String type = rdfClient.getTimeSeriesType(tsIRI);
+		Iri timeSeriesType = null;
+		if(type.equals(TimeSeriesSparql.Average+"TimeSeries")){
+			timeSeriesType = TimeSeriesSparql.AverageTimeSeries;
+		}
+		else if(type.equals(TimeSeriesSparql.Instantaneous+"TimeSeries")){
+			timeSeriesType = TimeSeriesSparql.InstantaneousTimeSeries;
+		}
+		else if(type.equals(TimeSeriesSparql.StepwiseCumulative+"TimeSeries")){
+			timeSeriesType = TimeSeriesSparql.StepwiseCumulativeTimeSeries;
+		}
+		else if(type.equals(TimeSeriesSparql.CumulativeTotal+"TimeSeries")){
+			timeSeriesType = TimeSeriesSparql.CumulativeTotalTimeSeries;
+		}
+		else if(type.equals(TimeSeriesSparql.ns_ontology+"TimeSeries")){
+			timeSeriesType = TimeSeriesSparql.TimeSeries;
+		}
```

Comment:
It's highly unlikely for this to be an issue, but there is potential for timeSeriesType to be null after this

Reply:
Maybe the time series triples were instantiated with an older version of the base lib and if the namespace is updated, this will fail

Reply:
the work around could be 

else { timeSeriesType = TimeSeriesSparql.TimeSeries; }

this would ensure if none of the types match, then it has to be the general time series type

Reply:
Not necessarily, if the namespace is different from the current version none will match, I suggest throwing an error